environment.yml

```diff
@@ -14,7 +14,8 @@ dependencies:
   - lightning~=2.1.0
   - matplotlib-base~=3.8.2
   - planetary-computer~=1.0.0
-  - pytorch~=2.1.0=*cuda120*
+  - pytorch~=2.1.0
+  - pyarrow~=15.0.0
```

Comment:
Could you resolve the merge conflict here with the `main` branch? Also see if removing the `pyarrow` pin works. I managed to get conda to solve for `osx-arm64` recently on https://github.com/Clay-foundation/model/pull/164 after https://github.com/conda-forge/torchvision-feedstock/pull/89/files was merged.

Reply:
Agreed. You should be able to simply ditch your changes to `environment.yml`. If not, then keep in mind that whenever you modify `environment.yml`, you must regenerate `conda-lock.yml`, so you should never be committing changes to _only_ `environment.yml`.

Reply:
Did you mean conflicts with [ci/osx-arm64](https://github.com/Clay-foundation/model/tree/ci/osx-arm64), rather than conflicts with `main`? I didn't notice any conflicts with `main`. 

I merged the changes from `ci/osx-arm64` and, while I'm able to install from `conda-lock.yml` (as @chuckwondo suggested) I'm not able to install from `environment.yml` due to `cuda` being unavailable in the docker container: 
```
> [claymodel 4/4] RUN micromamba create -y -n claymodel --file environment.yml &&     micromamba clean --all --yes:
135.1 error    libmamba Could not solve for environment specs
135.1     The following package could not be installed
135.1     └─ pytorch ~=2.1.0 *cuda12* is not installable because it requires
135.1        └─ __cuda, which is missing on the system.
135.3 critical libmamba Could not solve for environment specs
------
failed to solve: process "/usr/local/bin/_dockerfile_shell.sh micromamba create -y -n claymodel --file environment.yml &&     micromamba clean --all --yes" did not complete successfully: exit code: 1
```

I'm thinking that the `conda-lock.yml` and `environment.yml` files aren't synchronized? 

Let me know what the best course of action is. I can look into making a [composite docker image](https://micromamba-docker.readthedocs.io/en/latest/advanced_usage.html#adding-micromamba-to-an-existing-docker-image), based off of both `micromamba` and [nvidia/cuda](https://hub.docker.com/r/nvidia/cuda) so that we can install pytorch with cuda support in the docker container

Reply:
Hm that's odd - there were no merge conflict with `main` in the case. I'm might have been missing something. Do the environment.yml and conda-lock.yml files look correct in their current state?